float64[] x_states
float64[] y_states
float64[] yaw_states
float64[] x_controls
float64[] y_controls
float64[] yaw_controls
float64[] control_durations